storage/src/backend/registry.rs

```diff
@@ -166,6 +186,16 @@ struct RegistryState {
     cached_bearer_auth: ArcSwapOption<BearerAuth>,
 }
 
+fn needs_fallback_http(e: &dyn Error) -> bool {
+    match e.source() {
+        Some(err) => match err.source() {
+            Some(err) => err.to_string().contains("error:1408F10B"),
```

Comment:
Not sure the error code stamp is compatible with all registry servers, I will test it on my machine, thanks!